File: src/ExpenseManagement.Domain/Entities/Budget.cs
using ExpenseManagement.Domain.Common;
using ExpenseManagement.Domain.Enums;

namespace ExpenseManagement.Domain.Entities;

public class Budget : AuditableEntity
{
    public string Name { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public decimal SpentAmount { get; set; }
    public BudgetPeriod Period { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public decimal WarningThreshold { get; set; } = 80; // Percentage
    public bool IsActive { get; set; }
    public string UserId { get; set; } = string.Empty;
    public Guid CategoryId { get; set; }

    // Navigation properties
    public Category Category { get; set; } = null!;
}

File: expense-management-ui/src/types/budget.ts
export enum BudgetPeriod {
  Monthly = 1,
  Quarterly = 2,
  Yearly = 3,
}

export interface Budget {
  id: string;
  name: string;
  amount: number;
  spentAmount: number;
  remainingAmount: number;
  percentage: number;
  period: BudgetPeriod;
  startDate: string;
  endDate: string;
  warningThreshold: number;
  isActive: boolean;
  categoryId: string;
  categoryName: string;
  categoryColor: string;
  isExceeded: boolean;
  isWarning: boolean;
}

export interface CreateBudgetDto {
  name: string;
  amount: number;
  period: BudgetPeriod;
  startDate: string;
  warningThreshold: number;
  categoryId: string;
}
